Lib/sndhdr.py

```diff
@@ -49,10 +49,16 @@
 SndHeaders.sampwidth.__doc__ = ("""Either the sample size in bits or
 'A' for A-LAW or 'U' for u-LAW.""")
 
+
 def what(filename):
     """Guess the type of a sound file."""
-    res = whathdr(filename)
-    return res
+    try:
+        res = whathdr(filename)
+        return res
+    except RuntimeError as e:
```

Comment:
>Should we only handle the specific exceptions that are created in these two examples? or is there a cleaner, more pythonic way to handle it?

I think this should follow the pattern set out by `sndhdr` and have the exception catching be on a per-test basis. Take a look at this block for example: https://github.com/python/cpython/blob/feabae961707b00008c15a31352e458f4e8b3a6c/Lib/sndhdr.py#L75-L91

You should figure out what exceptions the individual tests code can throw and catch those specifically. As it stands now, its possible for a coding error like an out of bounds exception to be caught accidentally by your exception handler.

Reply:
cool, in that case we catch exactly those two errors, the `RuntimeError` and the `IndexError` caused by these bad files and catch them explicitly. I'll add some testcases to `Lib/test_sndhdr.py` and @you when it's done.